src/workers/postgres.rs

```diff
@@ -23,36 +23,75 @@ use crate::encoding::BinaryEncoding;
 
 lazy_static::lazy_static! {
     pub static ref MESSAGES_IN_POSTGRES_CHANNEL: GenericGauge<prometheus::core::AtomicI64> = register_int_gauge!(opts!("literpc_messages_in_postgres", "Number of messages in postgres")).unwrap();
+    pub static ref POSTGRES_SESSION_ERRORS: GenericGauge<prometheus::core::AtomicI64> = register_int_gauge!(opts!("literpc_session_errors", "Number of failures while establishing postgres session")).unwrap();
+}
+
+const MAX_QUERY_SIZE: usize = 200_000; // 0.2 mb
+
+trait SchemaSize {
+    const DEFAULT_SIZE: usize = 1;
+    const MAX_SIZE: usize = 1;
+}
+
+const fn get_max_safe_inserts<T: SchemaSize>() -> usize {
+    if T::DEFAULT_SIZE == 0 {
+        panic!("DEFAULT_SIZE can't be 0. SchemaSize impl should override the DEFAULT_SIZE const");
+    }
+
+    MAX_QUERY_SIZE / T::DEFAULT_SIZE
+}
+
+const fn get_max_safe_updates<T: SchemaSize>() -> usize {
+    if T::MAX_SIZE == 0 {
+        panic!("MAX_SIZE can't be 0. SchemaSize impl should override the MAX_SIZE const");
+    }
+
+    MAX_QUERY_SIZE / T::MAX_SIZE
 }
 
 #[derive(Debug)]
 pub struct PostgresTx {
-    pub signature: String,
-    pub recent_slot: i64,
-    pub forwarded_slot: i64,
-    pub forwarded_local_time: DateTime<Utc>,
+    pub signature: String,                   // 88 bytes
+    pub recent_slot: i64,                    // 8 bytes
+    pub forwarded_slot: i64,                 // 8 bytes
+    pub forwarded_local_time: DateTime<Utc>, // 8 bytes
     pub processed_slot: Option<i64>,
```

Comment:
Option should be 1 bytes or 9 bytes right?

Reply:
It is not sent in the query. I am not calculating struct size, but the size of the actual query